streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBuffer.java

```diff
@@ -0,0 +1,188 @@
+                    );
+                }
+                callback.accept(new Eviction<>(key, value, bufferValue.context()));
+                wrapped().remove(keyValue.key);
+                numRec--;
+                bufferSize = bufferSize - computeRecordSize(keyValue.key, bufferValue);
+                if (iterator.hasNext()) {
+                    keyValue = iterator.next();
+                    if (keyValue == null) {
+                        minTimestamp = Long.MAX_VALUE;
+                    } else {
+                        bufferValue = BufferValue.deserialize(ByteBuffer.wrap(keyValue.value));
+                        key = keySerde.deserializer().deserialize(topic,
+                            PrefixedWindowKeySchemas.TimeFirstWindowKeySchema.extractStoreKeyBytes(keyValue.key.get()));
+                        value = valueSerde.deserializeParts(
+                            topic,
+                            new Change<>(bufferValue.newValue(), bufferValue.oldValue())
+                        );
+                        minTimestamp = bufferValue.context().timestamp();
+                    }
+                } else {
+                    keyValue = null;
+                    minTimestamp = Long.MAX_VALUE;
+                }
+            }
+        }
+    }
+
+
+    @Override
+    public Maybe<ValueAndTimestamp<V>> priorValueForBuffered(final K key) {
+        return null;
+    }
+
+    @Override
+    public void put(final long time, final Record<K, Change<V>> record, final ProcessorRecordContext recordContext) {
+        requireNonNull(record.value(), "value cannot be null");
+        requireNonNull(recordContext, "recordContext cannot be null");
+        final Bytes serializedKey = Bytes.wrap(
+            PrefixedWindowKeySchemas.TimeFirstWindowKeySchema.toStoreKeyBinary(keySerde.serializer().serialize(topic, record.key()),
+                record.timestamp(),
+                0).get());
+        final Change<byte[]> serialChange = valueSerde.serializeParts(topic, record.value());
+        final BufferValue buffered = new BufferValue(serialChange.oldValue, serialChange.oldValue, serialChange.newValue, recordContext);
+        if (wrapped().observedStreamTime - gracePeriod.toMillis() > record.timestamp()) {
+            return;
+        }
+        wrapped().put(serializedKey, buffered.serialize(0).array());
+        bufferSize += computeRecordSize(serializedKey, buffered);
+        numRec++;
+        if (minTimestamp() > record.timestamp()) {
+            minTimestamp = record.timestamp();
+        }
+    }
+
+    @Override
+    public int numRecords() {
+        return numRec;
+    }
+
+    @Override
+    public long bufferSize() {
+        return bufferSize;
+    }
+
+    @Override
+    public long minTimestamp() {
+        return minTimestamp;
+    }
+
+    private static long computeRecordSize(final Bytes key, final BufferValue value) {
+        long size = 0L;
+        size += 8; // buffer time
```

Comment:
Does this comment refer to the size of the record timestamp that's serialized with the record key? If so, I think that's double-counted now, since the key passed to this method is already the serialized form, which includes the timestamp.